app/javascript/utils/url-params.ts

```diff
@@ -1,12 +1,18 @@
+type Params = {
+  [key: string]: any
+}
+
```

Comment:
It's easier to write

```suggestion
type Params = Record<string, any>

```

Are you certain it can be _any_ tho? From line 18-25, the reality is it can only be an `Array` of `string` (`string[]`), or whatever `encodeURIComponent` takes (which is probably only `string`).

```suggestion
type Params = Record<string, string | string[]>
```

Reply:
I've refined this is bit